### PR TITLE
Nodeaffinity vk

### PR DESCRIPTION
**Pull request by nilekhc**
**Reason for Change**:

Change adds nodeaffinity to prevent scheduling on virtual-kubelet nodes.

**Requirements**

- [ ] squashed commits
- [ ] included documentation
- [ ] added unit tests and e2e tests (if applicable).

**Issue Fixed**:

https://github.com/Azure/secrets-store-csi-driver-provider-azure/issues/472

**Does this change contain code from or inspired by another project?**

- [ ] Yes
- [x] No

**If "Yes," did you notify that project's maintainers and provide attribution?**

**Special Notes for Reviewers**:
Should we create new patch version(0.0.19) for chart?

### CODE REVIEW

**Review by aramase — COMMENTED** [review]
Thank you for looking into this. The changes look good, just one small nit:

1. For chart or deployment yaml updates we update the yamls in the [manifest_staging/](https://github.com/Azure/secrets-store-csi-driver-provider-azure/tree/master/manifest_staging/) folder, where we host the staging charts and deployment yamls. All the yaml changes will then be promoted into the released charts folder with the next release. 
2. Also could you add the new configurable values to the configuration [table](https://github.com/Azure/secrets-store-csi-driver-provider-azure/tree/master/manifest_staging/charts/csi-secrets-store-provider-azure#configuration).

**Comment by nilekhc** [comment]
Thanks for the review @aramase .  I have addressed above comments. Also added node affinity to deployment yamls.

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 1]
```diff
@@ -72,6 +72,7 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
 | `linux.volumes`                                                  | Additional volumes to create for the KeyVault provider pods.                                                                                                                                          | `[]`                                                                                             |
 | `linux.volumeMounts`                                             | Additional volumes to mount on the KeyVault provider pods.                                                                                                                                            | `[]`                                                                                             |
+| `linux.affinity`                                                 | Cofigures node affinity for provider pods on linux nodes                   | Match expression `type NotIn virtual-kubelet`                                                                              |
```
[the remark this answers is not in the document]
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/README.md]: updated

File: manifest_staging/charts/csi-secrets-store-provider-azure/README.md [hunk 2]
```diff
@@ -88,6 +89,7 @@ The following table lists the configurable parameters of the csi-secrets-store-p
...
 | `windows.healthzTimeout`                                         | RPC timeout for health check                                                                                                                                                                          | `"5s"`                                                                                           |
 | `windows.volumes`                                                | Additional volumes to create for the KeyVault provider pods.                                                                                                                                          | `[]`                                                                                             |
+| `windows.affinity`                                                 | Cofigures node affinity for provider pods on windows nodes                   | Match expression `type NotIn virtual-kubelet`                                                                              |
```
[the remark this answers is not in the document]
Comment (nilekhc) [reply manifest_staging/charts/csi-secrets-store-provider-azure/README.md]: updated

**Review by aramase — APPROVED** [review]
lgtm

**Pull request merged**